src/main/java/com/basho/riak/client/api/commands/datatypes/GSetUpdate.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2016 Basho Technologies Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.basho.riak.client.api.commands.datatypes;
+
+import com.basho.riak.client.core.query.crdt.ops.GSetOp;
+import com.basho.riak.client.core.query.crdt.ops.SetOp;
+import com.basho.riak.client.core.util.BinaryValue;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * An update to a Riak grow-only set datatype.
+ * <p>
+ * When building an {@link UpdateSet} command
+ * this class is used to encapsulate the update to be performed on a
+ * Riak gset datatype.
+ * </p>
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.2
+ */
+public class GSetUpdate implements DatatypeUpdate
+{
+    protected final Set<BinaryValue> adds = new HashSet<>();
```

Comment:
I don't think it's necessary to cache it, if somebody is breaking the immutability of BinaryValue then we don't have any guarantees for them or the API.